sklearn/mixture/tests/test_gaussian_mixture_ic.py

```diff
@@ -0,0 +1,264 @@
+from numpy.testing import assert_allclose, assert_equal
+from sklearn.exceptions import NotFittedError
+from sklearn.metrics import adjusted_rand_score
+
+from sklearn.mixture import GaussianMixtureIC
+
+
+def _test_inputs(X, error_type, **kws):
+    with pytest.raises(error_type):
+        gmIC = GaussianMixtureIC(**kws)
+        gmIC.fit(X)
+
+
+def test_n_components():
+    # Generate random data
+    X = np.random.normal(0, 1, size=(100, 3))
+
+    # min_components < 1
+    _test_inputs(X, ValueError, min_components=0)
+
+    # min_components integer
+    _test_inputs(X, TypeError, min_components="1")
+
+    # max_components < min_components
+    _test_inputs(X, ValueError, max_components=0)
+
+    # max_components integer
+    _test_inputs(X, TypeError, max_components="1")
+
+    # max_components > n_samples
+    _test_inputs(X, ValueError, max_components=101)
+
+    # min_components > n_samples
+    _test_inputs(
+        X, ValueError, **{"min_components": 101, "max_components": 102}
+    )
+
+
+def test_input_param():
+    # Generate random data
+    X = np.random.normal(0, 1, size=(100, 3))
+
+    # affinity is not an array, string or list
+    _test_inputs(X, TypeError, affinity=1)
+
+    # affinity is not in ['euclidean', 'manhattan', 'cosine', 'none']
+    _test_inputs(X, ValueError, affinity="1")
+
+    # linkage is not an array, string or list
+    _test_inputs(X, TypeError, linkage=1)
+
+    # linkage is not in ['single', 'average', 'complete', 'ward']
+    _test_inputs(X, ValueError, linkage="1")
+
+    # covariance type is not an array, string or list
+    _test_inputs(X, TypeError, covariance_type=1)
+
+    # covariance type is not in ['spherical', 'diag', 'tied', 'full']
+    _test_inputs(X, ValueError, covariance_type="1")
+
+    # euclidean is not an affinity option when ward is a linkage option
+    _test_inputs(X, ValueError, **{"affinity": "manhattan", "linkage": "ward"})
+
+    # criterion is not "aic" or "bic"
+    _test_inputs(X, ValueError, criterion="cic")
+
+
+def test_labels_init():
+    # Generate random data
+    X = np.random.normal(0, 1, size=(100, 3))
+
+    # label_init is not a 1-D array
+    _test_inputs(X, TypeError, label_init=np.zeros([100, 2]))
+
+    # label_init is not 1-D array, a list or None.
+    _test_inputs(X, TypeError, label_init="label")
+
+    # label_init length is not equal to n_samples
+    _test_inputs(X, ValueError, label_init=np.zeros([50, 1]))
+
+    # label_init length does not match min_components and max_components
+    _test_inputs(
+        X,
+        ValueError,
+        **{
+            "label_init": np.zeros([100, 1]),
+            "min_components": 2,
+            "max_components": 3,
+        },
+    )
+
+
+def test_predict_without_fit():
+    # Generate random data
+    X = np.random.normal(0, 1, size=(100, 3))
+
+    with pytest.raises(NotFittedError):
+        gmIC = GaussianMixtureIC(min_components=2)
+        gmIC.predict(X)
+
+
+def test_cosine_with_0():
```

Comment:
Good call!

Reply:
Maybe add a small comment [to give the context](https://github.com/scikit-learn/scikit-learn/pull/19562/files#diff-6c6a458c6e3e062e9487cb1d917ff2841086fdf47abd9c3588095af0decc513bR69-R71) and the purpose of this test.